zerver/views/message_edit.py

```diff
@@ -180,4 +181,18 @@ def json_fetch_raw_message(
     else:
         (message, user_message) = access_message(maybe_user_profile, message_id)
 
-    return json_success({"raw_content": message.content})
+    # Return requested properties of message if user has access to the message.
+    if properties:
+        values = {}
+        for prop in properties:
+            if message.is_stream_message():
+                if prop == "topic":
+                    values[prop] = message.topic_name()
+                elif prop == "stream":
+                    values[prop] = Stream.objects.get(id=message.recipient.type_id).name
+
+        print(values)
+        return json_success(values)
+    else:
+        # Return message raw content which is used by message edit workflow.
+        return json_success({"raw_content": message.content})
```

Comment:
I think it might be better to just add fields to the payload unconditionally, since that'd result in a much cleaner API; it's very cheap to do so and would be more useful.  Maybe the cleanest would be to just have this return the full Message object in the same format that `GET /messages` does with its bulk queries, including the raw content, so that we don't have to do API changes whenever we come up with a new reason to want other fields here?

We can then make this a separate/preparatory API change PR that should be able to merge fairly quickly.